examples/ct/ct_server/main.go

```diff
@@ -43,34 +46,30 @@ type config struct {
 	LogConfig            string
 	MaxGetEntriesAllowed int64
 	EtcdServers          string
+
+	configs.GLogFlags
 }
 
-var configFlag = flag.String("config", "", "Path to JSON config file to load instead of providing config via flags, if provided may only be used along with the following flags: --logtostderr, --alsologtostderr, --stderrthreshold, --log_dir, --log_backtrace_at, --v, --vmodule")
+var configFlag = flag.String("config", "", "Path to JSON config file to load instead of providing config via flags, if provided no other flags may be used")
 
 func main() {
 	c := config{}
-	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
-	flags.StringVar(&c.Host, "host", "localhost", "Address to serve CT log requests on")
-	flags.IntVar(&c.Port, "port", 6962, "Port to serve CT log requests on")
-	flags.StringVar(&c.LogRPCServer, "log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
-	flags.DurationVar(&c.RPCDeadline.Duration, "rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
-	flags.StringVar(&c.LogConfig, "log_config", "", "File holding log config in JSON")
-	flags.Int64Var(&c.MaxGetEntriesAllowed, "maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
-	flags.StringVar(&c.EtcdServers, "etcd_servers", "", "A comma-separated list of etcd servers")
-	// Both the default flag set and the binary specific flag set need to have Parse() called
-	// so that we can also initialize the flags registered by github.com/golang/glog as well
-	// as the ones defined above
-	flags.Parse(os.Args[1:])
+	flag.StringVar(&c.Host, "host", "localhost", "Address to serve CT log requests on")
+	flag.IntVar(&c.Port, "port", 6962, "Port to serve CT log requests on")
+	flag.StringVar(&c.LogRPCServer, "log_rpc_server", "localhost:8090", "Backend specification; comma-separated list or etcd service name (if --etcd_servers specified)")
+	flag.DurationVar(&c.RPCDeadline.Duration, "rpc_deadline", time.Second*10, "Deadline for backend RPC requests")
+	flag.StringVar(&c.LogConfig, "log_config", "", "File holding log config in JSON")
+	flag.Int64Var(&c.MaxGetEntriesAllowed, "maxGetEntriesAllowed", 0, "Max number of entries we allow in a get-entries request (default 50)")
+	flag.StringVar(&c.EtcdServers, "etcd_servers", "", "A comma-separated list of etcd servers")
 	flag.Parse()
 
 	if *configFlag != "" {
-		// Only allow either --config OR any other flags
-		if flags.NFlag() > 1 {
-			glog.Exit("Cannot use --config with any other flags")
-		}
 		if err := configs.LoadConfig(*configFlag, &c); err != nil {
 			glog.Exit(err)
 		}
+		if err := configs.SetGLogFlags(c.GLogFlags); err != nil {
```

Comment:
Suggestion: Could we move this to LoadConfig?

If we make changes similar to the ones below I reckon it should work:

```go

// Change SetGLogFlags signature.
func (gf *GLogFlags) SetGLogFlags() error { ... }

type glogFlagsSetter interface {
    SetGLogFlags() error
}

func LoadConfig(path string, obj interface{}) error {
  // ...
  if setter, ok := obj.(glogFlagsSetter); ok {
    if err := setter.SetGlogFlags(); err != nil {
      return err
    }
  }
}
```